Acd 827 wfii (#88)
Actindo-AG/Documentation: ActindoWorkFlow/UserInterface/08_CoreActions.md

![Core actions](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/CoreActions.png "[Core actions]")

> To create a workflow, you must include one or several process actions.   

Reviewer: Reintheoretisch sollte es "To create a workflow, you must include one or several transitions" heißen, oder? Process action heißt es, wenn es zu den tatsächlich Prozessausführung kommt (so ist es vorher beschrieben, denke ich). Hier ist die Terminologie (transition vs. action) unglücklich.

### Configuration   

**Static inputs**

The *Change process priority* core action has two input ports:

- loop through: The data run through the transition without any changes.
- priority: Enter the priority. The input value of the *priority* input port must be an integer. This integer will be used as the new priority. 

## Multiply input action

![Multiply input action](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/MultiplyInputAction.png "[Multiply input action]")

The *Multiply input action* core action is used to output the data coming in via one input port to two output ports. This core action is often used when the same data is needed for two different purposes. To merge the duplicated data again, the *Wait for parallel input* core action can be used, see [Wait for parallel input](#wait-for-parallel-input).   
The data runs via the p input port into the workflow action and is output via both the p0 and the p1 output ports.

### Configuration 

This core action has no further configuration settings.

## Extract value

![Extract value action](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/ExtractValue.png "[Extract value action]")

> The *Extract value* core action is used to extract a specific value and to use it as input for further processing. For example, you extract a specific value from a business document position such as the *pimProduct* field of the **Actindo.Modules.RetailSuite.RetailSuiteFaktBase.Models.BusinessDocumentPosition** data model, or you extract a certain position of a business document. 

Reviewer: , such as (ich würde da ein Komma vor such as setzen)

You define the value to be extracted in the *Path to value* field in the configuration section. 

### Configuration

*Path to value*   
Search for the parameter name of the objects' data model that is available in the input port of the transition, copy the required parameter name, and enter it in the *Path to value* field, for example **pimProduct.pim.variants**.

## Manipulate process priority

![Manipulate process priority](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/ManipulateProcessPriority.png "[Manipulate process priority]")

> The *Manipulate process priority* core action is used to change the preconfigured priority of the process while the process is already running. In contrast to the [Change process priority](#change-process-priority) core action, here you define the new priority by a user input. This means that the priority is set statically and not dynamically as with the *Change process priority* transition.

Reviewer: Vorschlag: preconfigured process priority (so heißt ich auch das Feld im Trigger)

> [Info] The process priority can be defined in the workflow configuration by a trigger, see [Create a trigger](../Operation/02_ManageTriggers.md#create-a-trigger).

**Configuration**

*User input priority*   
Enter the absolute priority number or add the relative number to the existing process priority, for example, +2 (increase the existing priority, that is the process gets a higher priority) or -3 (decrease the existing priority, that is the process gets a lower priority).

## Manual action

![Manual action](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/ManualAction.png "[Manual action]")

> The *Manual action* core action is used to control the workflow by a manual user interaction. Depending on the user's decision made in the *Process orchestration >Processes > Process ID* view, a different workflow path may than be taken, see [Make a user decision](../Operation/09_TrackWorkflowProcess.md#make-a-user-decision). If a workflow contains a manual action, the corresponding process is paused until the user has done the decision.      

Reviewer: ```suggestion
The *Manual action* core action is used to control the workflow by a manual user interaction. Depending on the user's decision made in the *Process orchestration > Processes > Process ID* view, a different workflow path may then be taken, see [Make a user decision](../Operation/09_TrackWorkflowProcess.md#make-a-user-decision). If a workflow contains a manual action, the corresponding process is paused until the user has made the decision.      
```

> For example, it allows you to incorporate approval procedures into your workflows, such as requiring any purchase order over a certain amount to be reviewed by a user before it can be released or canceled. For this example, you can use the [Split by criterion](#split-by-criterion) action before to define the business document total amount for which a manual action is required.   

Reviewer: Vorschlag: In this case, you can use... (um example nicht zu wiederholen)

**Static inputs**   
   
The *Manual action* core action has two input ports:
- data_input: Entity that can be used to make a decision, such as a business document. 
- event-id: ID of the task event that can be used to make a decision.   

## Execute PHP code

![Execute PHP code](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/ExecutePHPCode.png "[Execute PHP code]")

The *Execute PHP code* core action is used to execute a custom PHP code defined in the configuration.
As the complete functionality of the PHP code is accessible, this core action enables complete variability in programming. Therefore, it is often used when more complex actions which require more logic need be executed within a workflow step.   
Depending on which ports are connected to the places and which ports are defined in the PHP code, the data runs via the *in0* to *in9* input ports into the process action and is output via the *out0* to *out9* output ports.   

> It is also possible to include a static input via the unconnected input ports. These inputs must be defined in the *Static inputs* section in the settings sidebar, see [Transitions](../Overview/04_WorkflowProcessElements#transitions).

Reviewer: ```suggestion
It is also possible to include a static input via the unconnected input ports. These inputs must be defined in the *Static inputs* section in the settings sidebar, see [Transitions](../Overview/04_WorkflowProcessElements.md#transitions).
```
Evtl. (auch) auf Insert static input verweisen?

![Split by criterion](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/SplitByCriterion.png "[Split by criterion]")

> The core action *Split by criterion* is used to compare the input value with a defined criterion and output it via a different branch depending on whether the input value matches or not.

Reviewer: ```suggestion
The *Split by criterion* core action is used to compare the input value with a defined criterion and output it via a different branch depending on whether the input value matches or not.
```

>         It is a match if the input value equals the defined value.

Reviewer: [Info] The properties of an entity you can refer to are described in the relevant API documentation. --> Vielleicht allgemeinen Pfad dahin hinzufügen? *Dev Tools > API > Module name > ...*

- *Value*   
    Enter the value to be compared with the input value. The value must be valid JSON and is type strict, for example a string must be specified as a string (**\"example\"**), an integer as an integer (**123**), and so on.

## Start subprocess

![Start subprocess](../../Assets/Screenshots/ActindoWorkFlow/Workflows/CoreActions/StartSubprocess.png "[Start subprocess]")

> The *Start subprocess* core action is used to start a different sub process within the currently running process.       

Reviewer: Ich finde es separat (sub process) schwieriger zu lesen, ehrlich gesagt. Merriam Webster sagt zusammen oder mit Bindestrich.
https://www.merriam-webster.com/dictionary/subprocess
In unserer Style Guide habe wir ähnliche Termini mit Bindestrich: sub-tab, sub-procedure... 
https://actindo.atlassian.net/wiki/spaces/DOC/pages/110886966/Terminology+for+Terminologue
Also lieber mit Bindestrich?
